### ISSUE TITLE
Long press delete by word

### ISSUE THREAD

**Issue by jeroen7s** [dslul/openboard#28]
Add an option to delete per word when long pressing delete.

**Comment by DonnieWest** [comment]
I've also missed this feature! I've maintained a little patch to AOSP based keyboards to give myself something similar

Here they are on this repo: https://github.com/DonnieWest/openboard/tree/swiftkey-faker

Maintaining my fork hasn't been a huge hassle, so I've not bothered to put in the footwork necessary to make this a toggle-able setting. Someone else is welcome to do so if they want

**Comment by jwsp1** [comment]
> I've also missed this feature! I've maintained a little patch to AOSP based keyboards to give myself something similar
> 
> Here they are on this repo: https://github.com/DonnieWest/openboard/tree/swiftkey-faker
> 
> Maintaining my fork hasn't been a huge hassle, so I've not bothered to put in the footwork necessary to make this a toggle-able setting. Someone else is welcome to do so if they want

Why does @dslul not simply include your patch? You can make pull requests. 
But are you planning to add more swiftkey like stuff? Maybe you both should help each other out instead of creating two different apps 😊

**Comment by DonnieWest** [comment]
Because my patch changes the behavior for everyone without exception and doesn't include the ability to conditionally select the behavior. I'm not distributing this as my own app, I'm just keeping it on my personal phone since I prefer the Swiftkey layout and long-press delete

The beauty of OSS though is that someone else is welcome to take my patch, fill in the missing functionality, and submit it as their own PR. I'd even be happy to walk them through it if they have any questions

**Comment by jwsp1** [comment]
> Because my patch changes the behavior for everyone without exception and doesn't include the ability to conditionally select the behavior. I'm not distributing this as my own app, I'm just keeping it on my personal phone since I prefer the Swiftkey layout and long-press delete
> 
> The beauty of OSS though is that someone else is welcome to take my patch, fill in the missing functionality, and submit it as their own PR. I'd even be happy to walk them through it if they have any questions

Ahh, I understand. Personally, I also prefer SwiftKey layout. I'm excited to see how the projects grow in future :)

**Comment by dslul** [comment]
@DonnieWest pull requests are always welcome, I can add the toggle afterwards :)

**Comment by GambaJo** [comment]
I would like to have thos option too.

**Comment by DonnieWest** [comment]
I'll package this up and a few other tweaks I've been meaning to do and submit a PR. Not sure when I'll get around to it

**Comment by dslul** [comment]
> I'll package this up and a few other tweaks I've been meaning to do and submit a PR. Not sure when I'll get around to it

great! I added a toggle to disable the globe button in the last update, so that part is no longer necessary.

**Comment by jwsp1** [comment]
> I'll package this up and a few other tweaks I've been meaning to do and submit a PR. Not sure when I'll get around to it

Awesome 😍

**Comment by robsmith11** [comment]
@DonnieWest is there a working version of your patch?

This is the only feature I'm missing from Swiftkey.

**Comment by DonnieWest** [comment]
No, there's no up to date patch and I'm unsure how much drift there's been in the codebase.

It's a pretty naive implementation and I both switched keyboards and realized it probably needed more work to be viable. Feel free to adapt it if you can or want.

https://github.com/openboard-team/openboard/commit/77e6e2d57cef386e7874f467dd95cdcc669d7873
